app/controllers/archived_conversations_controller.rb

```diff
@@ -0,0 +1,37 @@
+class ArchivedConversationsController < ApplicationController
+  before_action :authenticate_user!
+
+  def index
+    @conversations = load_archived
+  end
+
+  def update
+    @conversation ||= Conversation.find(params[:id])
+
+    # update the unarchived conversation to archive of a user type
+    if @conversation.unarchived_by?(current_user)
+      @conversation.archive_from(user_type)
+      load_all
+      redirect_to conversations_path
+    else
+      @conversation.update("#{user_type}_archived_at": nil)
+      @conversations = load_archived
+      redirect_to archived_conversations_path
+    end
+  end
+
+  private
+
+  def load_all
+    @conversations = current_user.conversations.send("unarchived_by_#{user_type}")
+    @archived_any = current_user.conversations.send("archived_by_#{user_type}").exists?
+  end
+
+  def load_archived
+    current_user.conversations.send("archived_by_#{user_type}")
+  end
+
+  def user_type
+    current_user.business ? "business" : "developer"
```

Comment:
This logic isn't perfect. A user could have a business _and_ a developer but be messaging from their developer "account".

Instead, we can use `conversation#business?(current_user)` and its pair to determine which field to update.

Reply:
Or check out `BlocksController`!

Reply:
Ok so the workaround for this as you said is to use the `business?` or `developer?` method to know the type of user which is accessible by `@conversation`. Great! But I will not be able to use that method on `@conversations` or in the `index` action! So I created this logic:

```
  def user_type
    if current_user.business
      :business_archived_at
    elsif current_user.developer
      :developer_archived_at
    end
  end
```
Is this something you were looking for?

Reply:
Oh good point. Unfortunately that logic still has the same flaw as before - a user can have a business but be involved in a conversation from the developer’s side.

Reply:
Exactly! It's exactly the same logic as before but with a different return value.

I gave a little thought to it & I came up with this logic inside the `index` action of `ConversationsController`:
```
  def index
    if current_user.business
      @conversations = current_user.business.conversations.unarchived_by_user(:business_archived_at)
      @archived_any = current_user.business.conversations.archived_by_user(:business_archived_at).exists?
    else
      @conversations = current_user.developer.conversations.unarchived_by_user(:developer_archived_at)
      @archived_any = current_user.developer.conversations.archived_by_user(:developer_archived_at).exists?
    end
  end

Reply:
But if a user has both they would never see their developer conversations. 

I recognize this is a problem in the existing logic of the app. User vs. Accounts and all that. I’m not sure the best approach just yet.

Reply:
Ohh! So If I am understanding this correctly, assuming a user has both a `developer` & `business` profiles, do you want to load all the unarchived conversations in `/conversation` for both developer & business scopes right? And the same for the archived conversation?

Reply:
Okk! I am not sure I am still following the details here! So I want to understand this step by step:

> For example, User#archived_conversations should return conversations where the user's developer profile has archived something or the user's business has archived something.

What if the user is both a developer & business? Would you like to show archived conversations for both the developer & business in `/conversations/archived` page?

Reply:
Yes. `/conversations` should show all non-archived conversations for the user. And `/conversations/archived` should show all archived conversations for the user. Both regardless of which, if either/any, business/developer profile they have.